tail/tail.go

```diff
@@ -118,7 +180,28 @@ func Tail(ctx context.Context, logger log.Logger, dir string, promMon *prometheu
 		logger:  logger,
 		monitor: promMon,
 	}
-	cpdir, k, err := wal.LastCheckpoint(dir)
+
```

Comment:
I would like to follow a different pattern for this timeout, which is followed _mostly_ consistently in the codebase, so that the SIGTERM signal will gracefully exit the sidecar, since `t.ctx` comes from `main.go` where the signal handler lives.

The pattern is to use a 

```
	ctx, cancel := context.WithTimeout(t.ctx, TIMEOUT)
	defer cancel()
``` 

outside the loop, then inside the loop

```
    		select {
		case <-ctx.Done():
		         return nil, errors.Wrap(ctx.Err(), "checkpoint in progress")
		default:
		}
```
(I think, then, you can remove `ErrCheckpointInProgress`).

Reply:
Done